pkgs/applications/terminal-emulators/mlterm/default.nix

```diff
@@ -1,14 +1,82 @@
-{ stdenv, lib, fetchFromGitHub, pkg-config, autoconf, makeDesktopItem, nixosTests
-, libX11, gdk-pixbuf, cairo, libXft, gtk3, vte
-, harfbuzz #substituting glyphs with opentype fonts
-, fribidi, m17n_lib #bidi and encoding
+{ stdenv
+, lib
+, fetchFromGitHub
+, pkg-config
+, autoconf
+, makeDesktopItem
+, nixosTests
+, vte
+# Open Type layout support, (substituting glyphs with opentype fonts)
+, enableOtl ? true
+, harfbuzz # can be replaced with libotf
+, enableBidi ? true
+, fribidi
+, enableM17n ? !stdenv.isDarwin
+, m17n_lib
+, enableSsh2 ? true
 , libssh2 #build-in ssh
-, fcitx5, fcitx5-gtk, ibus, uim #IME
+, enableFcitx ? !stdenv.isDarwin
+, fcitx5, fcitx5-gtk
+, enableIbus ? !stdenv.isDarwin
+, ibus
+, enableUim ? !stdenv.isDarwin
+, uim #IME
 , wrapGAppsHook #color picker in mlconfig
-, Cocoa #Darwin
+, imagelibChoice ? "gdk-pixbuf" # or mlimgloader depending on your bugs of choice
+, gdk-pixbuf
+, gtk3
+, gtk ? gtk3
+# List of gui libraries to use. According to `./configure --help` ran on
+# release 3.9.3, options are: (xlib|win32|fb|quartz|console|wayland|sdl2|beos)
+, guiChoices ? [
+  "xlib"
+  "fb"
+]
+, xquartz # for the "quartz" --with-gui option
+, libxkbcommon , wayland # for the "wayland" --with-gui option
+, SDL2 # for the "sdl" --with-gui option
+# List of typing engines, the default list enables compiling all of the
+# available ones, as recorded on release 3.9.3
+, typeEnginesChoices ? [
+  "xcore"
+  "xft"
+  "cairo"
+]
+, libX11
+, libXft
+, cairo
+# List of external tools to create, this default list includes all default
+# tools, as recorded on release 3.9.3. mlterm-menu and mlconfig depend on
+# enabling gnome.at-spi2-core and configuring ~/.mlterm/key correctly.
+, toolsChoices ? [
```

Comment:
Add `assert`  guard code for this lists. (use cmake expression as an example)

Reply:
> Add `assert` guard code for this lists. (use cmake expression as an example)

TBH, here on Gnome I haven't experienced that this dependency is required, and I didn't understand why it didn't appear in the original expression, from which I only inherited this comment. Maybe I'll just remove it, and let users in the future add that dependency and assert if needed. Unless you can confirm this dependency is required.